tests/callbacks/test_lr_monitor.py

```diff
@@ -283,6 +283,77 @@ def configure_optimizers(self):
     assert lr_monitor.lr_sch_names == list(lr_monitor.lrs.keys()) == ['my_logging_name']
 
 
+def test_lr_monitor_custom_pg_name(tmpdir):
+
+    class TestModel(BoringModel):
+
+        def configure_optimizers(self):
+            optimizer = torch.optim.SGD([{'params': [p for p in self.layer.parameters()], 'name': 'linear'}], lr=0.1)
+            lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
+            return [optimizer], [lr_scheduler]
+
+    lr_monitor = LearningRateMonitor()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=2,
+        limit_val_batches=0.1,
+        limit_train_batches=0.5,
```

Comment:
This will run 10% and 50% of the available batches respectively, do we need that many?

You can use `int` to set a specific number of batches

Same for the test below

Reply:
Makes sense, these were clearly copy/paste UTs ;)

Reply:
Significant tests speedup! Excellent suggestion!